core/pkg/server/server.go

```diff
@@ -66,7 +85,23 @@ func (s *Server) SetDefaultLoggerPath(path string) {
 
 // Serve starts the server
 func (s *Server) Start() {
+	// add reference for main server loop
 	s.wg.Add(1)
+
+	// watch for parent process exit in background (if specified)
+	if s.pid != 0 {
+		s.wg.Add(1)
+		go func() {
+			shouldExit := s.loopCheckIfParentGone(s.pid)
+			if shouldExit {
+				slog.Info("Parent process exited, terminating core process")
+				os.Exit(1)
+			}
+			s.wg.Done()
+		}()
+	}
+
+	// run server in background
```

Comment:
let's move the wait group counter closer here instead of the top?

Reply:
sure, it was a bit more future proof by putting it earlier (if the loop check goroutine wants to be more graceful instead of an exit)... but given the current codebase it is fine to have where you recommend.